crypto/src/keys/ivk.rs

```diff
@@ -54,7 +54,8 @@ impl IncomingViewingKey {
         while u128::from_le_bytes(random_index) <= 2u128.pow(64) {
             rng.fill_bytes(&mut random_index);
         }
-        let index = AddressIndex::Random(random_index);
+        let index = AddressIndex::try_from(random_index.as_slice())
+            .expect("can form address index from slice");
```

Comment:
We don't want to do this, now that we changed the semantics of how ephemeral addresses work — the ephemeral_address function should take an account index as an input, rather than generating a random address for a random account.